kmk/modules/split.py

```diff
@@ -209,14 +215,40 @@ def on_powersave_disable(self, keyboard):
                 self._uart_connection.connection_interval = 11.25
                 self._psave_enable = False
 
-    def _check_all_connections(self):
+    def _check_all_connections(self, keyboard):
         '''Validates the correct number of BLE connections'''
+        self._previous_connection_count = self._connection_count
         self._connection_count = len(self._ble.connections)
-        if self._is_target and self._connection_count < 2:
-            self._target_advertise()
+        if self._is_target:
+            if self._advertising or not self._check_if_split_connected():
+                self._target_advertise()
+            elif self._connection_count < 2 and keyboard.hid_type == HIDModes.BLE:
+                keyboard._hid_helper.start_advertising()
```

Comment:
I can't see that messages even if I set `keyboard.go(debug_enabled=True)` ;<

I dislike restart advertising, so I suggest change [BLEHID.start_advertising()](https://github.com/yswallow/kmk_firmware/blob/eadbae3b1cb67c4dc53949d8323160ac663a2c20/kmk/hid.py#L320) as follows:

```python
self.ble.start_advertising(advertisement)
```

to

```python
if not self.ble.advertising:
    self.ble.start_advertising(advertisement)
```

Reply:
You're right. I retested it your way and it still works. I thought the issue was with the ongoing advertising which needed to be stopped, but it seems it wasn't that. If so, I'm not sure if we should include `ble_refresh` in `handlers/stock.py` in that way. Maybe we should delete `stop_advertising()` from there, maybe also rename function to `ble_advertise`.

Reply:
I think current implementation of `ble_refresh` is correct, because a keyboard already advertises UART for connect to right side must stop advertising before start advertising HID. And user won't press `BLE_REFRESH` key so often, restart advertising won't be an issue.